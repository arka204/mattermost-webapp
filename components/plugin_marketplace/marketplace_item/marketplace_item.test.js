// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import MarketplaceItem from './marketplace_item.js';

describe('components/MarketplaceItem', () => {
    const baseProps = {
        id: 'id',
        name: 'name',
        description: 'test plugin',
        version: '1.0.0',
        downloadUrl: 'http://example.com/download',
        signatureUrl: 'http://example.com/signature',
        homepageUrl: 'http://example.com',
        iconUrl: '',
        installedVersion: '1.0.0',
        onConfigure: () => {}, // eslint-disable-line no-empty-function
        installing: false,
        actions: {
            installMarketplacePlugin: () => {}, // eslint-disable-line no-empty-function
        },
    };

    test('should match snapshot, no plugin icon', () => {
        const wrapper = shallow(
            <MarketplaceItem {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with plugin icon', () => {
        const props = {
            ...baseProps,
            iconData: 'icon',
        };
        const wrapper = shallow(
            <MarketplaceItem {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with homepage url', () => {
        const wrapper = shallow(
            <MarketplaceItem {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, no homepage url', () => {
        const props = {
            ...baseProps,
        };
        delete props.homepageUrl;

        const wrapper = shallow(
            <MarketplaceItem {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, with server error', () => {
        const props = {
            ...baseProps,
            error: 'An error occurred.',
        };

        const wrapper = shallow(
            <MarketplaceItem {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
