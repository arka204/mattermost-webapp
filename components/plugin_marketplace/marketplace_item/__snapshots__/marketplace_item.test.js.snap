// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MarketplaceItem should match snapshot, no homepage url 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <PluginIcon
      className="icon__plugin icon__plugin--background"
    />
    <div
      className="more-modal__details"
    >
      <span
        aria-label="name, test plugin"
        className="style--none"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </span>
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem should match snapshot, no plugin icon 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <PluginIcon
      className="icon__plugin icon__plugin--background"
    />
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem should match snapshot, with homepage url 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <PluginIcon
      className="icon__plugin icon__plugin--background"
    />
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem should match snapshot, with plugin icon 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link"
    id="marketplace-plugin-id"
    key="id"
  >
    <div
      className="icon__plugin icon__plugin--background"
    >
      <img
        src="icon"
      />
    </div>
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description"
        >
          test plugin
        </p>
      </a>
    </div>
    <div
      className="more-modal__actions"
    >
      <Link
        to="/admin_console/plugins/plugin_id"
      >
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Configure"
            id="marketplace_modal.list.configure"
            values={Object {}}
          />
        </button>
      </Link>
    </div>
  </div>
</Fragment>
`;

exports[`components/MarketplaceItem should match snapshot, with server error 1`] = `
<Fragment>
  <div
    className="more-modal__row more-modal__row--link item_error"
    id="marketplace-plugin-id"
    key="id"
  >
    <PluginIcon
      className="icon__plugin icon__plugin--background"
    />
    <div
      className="more-modal__details"
    >
      <a
        aria-label="name, test plugin"
        className="style--none more-modal__row--link"
        href="http://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        name
         
        <span
          className="light subtitle"
        >
          (1.0.0)
        </span>
        <p
          className="more-modal__description error_text"
        >
          An error occurred.
        </p>
      </a>
    </div>
    <div
      className="more-modal__actions"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        <LoadingWrapper
          loading={false}
          text="Installing..."
        >
          <FormattedMessage
            defaultMessage="Try Again"
            id="marketplace_modal.list.try_again"
            values={Object {}}
          />
        </LoadingWrapper>
      </button>
    </div>
  </div>
</Fragment>
`;
